python/pyspark/errors/error_classes.py

```diff
@@ -622,6 +622,11 @@
       "No active Spark session found. Please create a new Spark session before running the code."
     ]
   },
+  "NO_ACTIVE_OR_DEFAULT_SESSION" : {
```

Comment:
We already have `NO_ACTIVE_SESSION`. Maybe we can update the existing error class?

Reply:
How is it different from `NOT_STR`, `NOT_STR_OR_LIST_OF_RDD` and `NOT_STR_OR_STRUCT` cases?

Reply:
They are type related error classes, so we separated them to inform users all the exact possible types for each cases.

On my second thought, we can also separate for this case, nvm.